### ISSUE TITLE
The installation instructions for Debian/Ubuntu should install `curl` if it is not installed

### ISSUE THREAD

**Issue by stasadev** [ddev/ddev#5915]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Output of `ddev debug test`

<details><summary>Expand `ddev debug test` diagnostic information</summary>

```
[COPY-PASTE HERE THE OUTPUT OF `ddev debug test`]
```
</details>         

### Expected Behavior

`curl` should be installed, otherwise `/etc/apt/keyrings/ddev.gpg` will be empty.

TODO:

- [ ] add `sudo apt update && sudo apt install -y curl` to the Linux instructions
- [ ] update wsl2 scripts
- [ ] update https://ddev.com/get-started/

### Actual Behavior

Unable to use the existing GPG key with Ubuntu 24.04:

```
# Add DDEV’s GPG key to your keyring
sudo sh -c 'echo ""'
sudo install -m 0755 -d /etc/apt/keyrings
curl -fsSL https://pkg.ddev.com/apt/gpg.key | gpg --dearmor | sudo tee /etc/apt/keyrings/ddev.gpg > /dev/null
sudo chmod a+r /etc/apt/keyrings/ddev.gpg

# Add DDEV releases to your package repository
sudo sh -c 'echo ""'
echo "deb [signed-by=/etc/apt/keyrings/ddev.gpg] https://pkg.ddev.com/apt/ * *" | sudo tee /etc/apt/sources.list.d/ddev.list >/dev/null

# Update package information and install DDEV
sudo sh -c 'echo ""'
sudo apt update && sudo apt install -y ddev
[sudo] password for stas: 

Command 'curl' not found, but can be installed with:
sudo snap install curl  # version 8.1.2, or
sudo apt  install curl  # version 8.4.0-2ubuntu1
See 'snap info curl' for additional versions.
gpg: no valid OpenPGP data found.


Hit:1 http://ua.archive.ubuntu.com/ubuntu noble InRelease
Hit:2 http://ua.archive.ubuntu.com/ubuntu noble-updates InRelease              
Hit:3 http://ua.archive.ubuntu.com/ubuntu noble-backports InRelease            
Hit:4 http://security.ubuntu.com/ubuntu noble-security InRelease               
Get:5 https://pkg.ddev.com/apt * InRelease                          
Err:5 https://pkg.ddev.com/apt * InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY C16711FC1179C7CA
Reading package lists... Done
W: GPG error: https://pkg.ddev.com/apt * InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY C16711FC1179C7CA
E: The repository 'https://pkg.ddev.com/apt * InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

### Steps To Reproduce

Run https://ddev.readthedocs.io/en/stable/users/install/ddev-installation/#debianubuntu on bare minimum Ubuntu 24.04 (without `curl` available).

### Anything else?

<details><summary>New requirements for APT repository signing</summary>
---------- Forwarded message ---------
From: Julian Andres Klode <jak@debian.org>
Date: Wed, Feb 28, 2024 at 12:21 PM
Subject: New requirements for APT repository signing
To: <debian-devel-announce@lists.debian.org>

APT 2.7.13 just landed in unstable and with GnuPG 2.4.5 installed,
or 2.4.4 with a backport from the 2.4 branch, requires repositories
to be signed using one of

- RSA keys of at least 2048 bit
- Ed25519
- Ed448

Any other keys will cause warnings. These warnings will become
errors in March as we harden it up for the Ubuntu 24.04 release,
which was the main driver to do the change *now*.

If you operate third-party repositories using different key
algorithms, now is your time to migrate before you get hit
with an error.

For the Ubuntu perspective, feel free to check out the discourse
post:

https://discourse.ubuntu.com/t/new-requirements-for-apt-repository-signing-in-24-04/42854
</details>

**Comment by rfay** [comment]
I wasn't able to get 24.04 fully installed, but was able to run the live preview, and got the same results, key not respected.

**Comment by rfay** [comment]
Our big problem will be updating the key without inconveniencing people who are using the current key.

**Comment by rfay** [comment]
I installed latest Ubuntu 24.04 in WSL and:
1. The key and repository were OK. `sudo apt update` worked ok.
2. I was able to install ddev just fine
3. Repositories for docker-ce were *not* available, so I'm not sure we have to worry about this yet.

I was installing from cloud-images, https://cloud-images.ubuntu.com/noble/current/, which has a date of 2024-02-20, and I used the technique in https://askubuntu.com/questions/1501519/how-to-install-ubuntu-24-04-pre-release-on-wsl

I definitely think we have to monitor this.

**Comment by rfay** [comment]
We are currently signing with an rsa 4096-bit key, which should meet the requirements:
```
$ gpg /etc/apt/keyrings/ddev.gpg
gpg: WARNING: no command supplied.  Trying to guess what you mean ...
pub   rsa4096 2022-07-05 [SC]
      DBD730A3A581BDAC98BAF17FC16711FC1179C7CA
uid           DDEV (Signing key for apt/rpm repos) <randy@localdev.foundation>
sub   rsa4096 2022-07-05 [E]
```

**Comment by rfay** [comment]
I think we can close this, close if you agree @stasadev

**Comment by stasadev** [comment]
Agree, closing.

**Comment by stasadev** [comment]
I just now realized why I got this error on Ubuntu 24.04:

```
The following signatures couldn't be verified because the public key is not available: NO_PUBKEY C16711FC1179C7CA
```

It was because `curl` is not available on the bare Ubuntu installation:

```
# Add DDEV’s GPG key to your keyring
sudo sh -c 'echo ""'
sudo install -m 0755 -d /etc/apt/keyrings
curl -fsSL https://pkg.ddev.com/apt/gpg.key | gpg --dearmor | sudo tee /etc/apt/keyrings/ddev.gpg > /dev/null
sudo chmod a+r /etc/apt/keyrings/ddev.gpg

# Add DDEV releases to your package repository
sudo sh -c 'echo ""'
echo "deb [signed-by=/etc/apt/keyrings/ddev.gpg] https://pkg.ddev.com/apt/ * *" | sudo tee /etc/apt/sources.list.d/ddev.list >/dev/null

# Update package information and install DDEV
sudo sh -c 'echo ""'
sudo apt update && sudo apt install -y ddev
[sudo] password for stas: 

Command 'curl' not found, but can be installed with:
sudo snap install curl  # version 8.1.2, or
sudo apt  install curl  # version 8.4.0-2ubuntu1
See 'snap info curl' for additional versions.
gpg: no valid OpenPGP data found.


Hit:1 http://ua.archive.ubuntu.com/ubuntu noble InRelease
Hit:2 http://ua.archive.ubuntu.com/ubuntu noble-updates InRelease              
Hit:3 http://ua.archive.ubuntu.com/ubuntu noble-backports InRelease            
Hit:4 http://security.ubuntu.com/ubuntu noble-security InRelease               
Get:5 https://pkg.ddev.com/apt * InRelease                          
Err:5 https://pkg.ddev.com/apt * InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY C16711FC1179C7CA
Reading package lists... Done
W: GPG error: https://pkg.ddev.com/apt * InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY C16711FC1179C7CA
E: The repository 'https://pkg.ddev.com/apt * InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

But I simply didn't see it, and run `sudo apt update` after this.

The install script needs to be updated to make sure `curl` is available.

**Comment by rfay** [comment]
I confirm that even the Ubuntu 24.04 docker image doesn't have curl. Odd to leave it out. 

I guess we just have to add `sudo apt update && sudo apt install -y curl` to the Linux instructions; it won't break anything. 

And that should be added to the two WSL2 install scripts as well, in case the new Ubuntu 24.04 WSL2 has this problem. I doubt it will, because the WSL2 distro is bigger than a base install.
